sympy/functions/elementary/piecewise.py

```diff
@@ -237,6 +238,29 @@ def _eval_derivative(self, x):
     def _eval_evalf(self, prec):
         return self.func(*[(e._evalf(prec), c) for e, c in self.args])
 
+    def _eval_is_meromorphic(self, x, a):
+        a_real = a.is_real
+        # First, let's check if a is real. If not return ValueError.
+        if a_real is False:
+            raise ValueError("A piecewise function is defined only on the real line.")
+        if a_real is None:
+            return None
+
+        # Then, scan ExprCondPairs in the given order to find a piece that would contain a,
+        # possibly as a boundary point.
+        for e, c in self.args:
+            if c is True:
+                c = S.true
```

Comment:
I only now realized that the condition `c` will never be `True` because of these lines: https://github.com/sympy/sympy/blob/ae895046cd7d5fea6194d46cb51776413ae25f1c/sympy/functions/elementary/piecewise.py#L23-L24
Hence the two lines above are dead code and should be removed.

Reply:
Hi. This is now complete. There is a failing check, but it seems to be the same issue to all pull requests.